src/include/migraphx/op/random_uniform.hpp

```diff
@@ -75,7 +75,20 @@ struct random_uniform
 
         result.visit([&](auto output) {
             using type = typename decltype(output)::value_type;
-            if constexpr(std::is_integral<type>{})
+#ifdef _MSC_VER
+            // According to the C++ specification, the effect is undefined if the result type
+            // for the generator is not one of short, int, long, long long, unsigned short,
+            // unsigned int, unsigned long, or unsigned long long. See
+            // https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution.
+            if constexpr(std::is_same_v<type, unsigned char> || std::is_same_v<type, signed char>)
```

Comment:
This could be written as `std::is_integral<type>{} and sizeof(type) == 1`, which I think is simpler. 

If not, then it should be `std::is_same<type, unsigned char>{} or std::is_same<type, signed char>{}` to match the style used throughout migraphx.